packages/@lwc/engine-server/src/apis/render-component.ts

```diff
@@ -11,15 +11,34 @@ import {
     setElementProto,
     LightningElement,
 } from '@lwc/engine-core';
+import { isString, isFunction, isObject, isNull } from '@lwc/shared';
 
 import { renderer } from '../renderer';
 import { serializeElement } from '../serializer';
 
 export function renderComponent(
     tagName: string,
     Ctor: typeof LightningElement,
-    props: { [name: string]: any }
+    props: { [name: string]: any } = {}
```

Comment:
note: in the past, we usually avoid default value due to the transpilation. I suspect that on server side it will be ok... just curious.

Reply:
I think the typescript compiler got smarter over time and now the generated code for default params looks good enough: https://www.typescriptlang.org/play/index.html?target=1#code/GYVwdgxgLglg9mABMOcAUcAOsEEMA2iAvIgN4C+AlGQFCL2IQIDOc+ApgHT5wDmG2eGAKUa5IA.

Reply:
very well!